Formula/lmod.rb

```diff
@@ -0,0 +1,64 @@
+class Lmod < Formula
+  desc "Lua-based environment modules system to modify PATH variable"
+  homepage "https://www.tacc.utexas.edu/research-development/tacc-projects/lmod"
+  url "https://github.com/TACC/Lmod/archive/7.7.16.tar.gz"
+  sha256 "cf14183f9954bf0d4c22db4b7db3a6459c8cb607b9d1ff25e18e0989b48bf58b"
+
+  depends_on "lua"
+
+  resource "luafilesystem" do
+    url "https://github.com/keplerproject/luafilesystem/archive/v1_7_0_2.tar.gz"
+    sha256 "23b4883aeb4fb90b2d0f338659f33a631f9df7a7e67c54115775a77d4ac3cc59"
+  end
+
+  resource "luaposix" do
+    # Don't use v34.0.2, it's broken (at least on macOS)
+    url "https://github.com/luaposix/luaposix/archive/v34.0.1.tar.gz"
+    sha256 "43e7fb1fa4a8b905dd256703aa62116937be72d07d3a46b66a8e9b2264921129"
+  end
+
+  def install
+    luapath = libexec/"vendor"
+    ENV["LUA_PATH"] = "#{luapath}/share/lua/5.3/?.lua" \
+                      ";#{luapath}/share/lua/5.3/?/init.lua"
+    ENV["LUA_CPATH"] = "#{luapath}/lib/lua/5.3/?.so"
+
+    resources.each do |r|
+      r.stage do
+        system "luarocks", "make", "--tree=#{luapath}"
+      end
+    end
+
+    system "./configure",
+      "--disable-debug",
+      "--disable-dependency-tracking",
+      "--disable-silent-rules",
+      "--prefix=#{prefix}"
+    system "make", "install"
+
+    # Move prefix/lmod/VERSION/lmod/VERSION/* to prefix/lmod/VERSION/*
```

Comment:
Why does it install in `prefix/lmod/VERSION/lmod/VERSION`? Sounds like a bug…

Reply:
I don't think this is true anymore.

Reply:
This is due to lmod's odd install to `PREFIX/lmod/version`, with `PREFIX/lmod/lmod` symlink. It doesn't play well with brew's management of versions. This coping fixes that. A better fix might be a disable versioning choice in lmod configure, might be worth requesting.

Reply:
Fixed the issue with version by fixing the lua path. It reads in the version using the current working directory, which was missing from the path. Using a symbolic link instead now, and a "versioned" directory to reduce the `lmod` repetition (2x vs. 3x).